src/runtime-helpers/runtime-helpers.ts

```diff
@@ -213,6 +227,37 @@ module ScriptLab {
     function _isDialogApiSupported() {
         return (window as any).Office.context.requirements.isSetSupported('DialogApi');
     }
+
+    function _extractAndCacheAccessToken(message: string, clientId: string, resource: string): string {
+        const authResponseKeyValues = message.split('&');
+        const accessToken = authResponseKeyValues[0].substr('AUTH:access_token='.length);
+        const expiresIn = Number(authResponseKeyValues[1].substr('AUTH:expires_in='.length));
+        cachedAuthTokens[clientId + resource] = { token: accessToken, expiry: Date.now() + ((expiresIn - 60) * 1000) };
```

Comment:
Please explain the magic number "60".  E.g.,: `60 /* one minute of safety margin */` or whatever.
BTW, if it is in fact just one minute, I would make it 5 instead, just in case.  That's what Office.js helpers do, too.